app/controllers/admin/adoption_applications_controller.rb

```diff
@@ -0,0 +1,22 @@
+class Admin::AdoptionApplicationsController < ApplicationController
+
+    def show
```

Comment:
The idea you mentioned in the video about this action is a great one -- it would be better to break up this functionality into 2 actions

1) the show page, which displays a button to update the status of the ApplicationPet by approving or rejecting a pet
2) an update action that can receive the request from this button, update the ApplicationPet and then redirect back to this admin show page

The question of where that update action can live is a good discussion, too. We wouldn't want it to be an update in the PetsController or ApplicationsController because that is typically reserved for updating attributes on the Pet or Application object. What we're doing here is updating the joins table record, so this is a great use case for a **joins controller**. You can name it ApplicationPetsController or AdoptionApplicationPetsController, and it can have an update action where you change the status and then redirect to this admin page.